R/margins.R

```diff
@@ -66,145 +64,80 @@ title_spec <- function(label, x, y, hjust, vjust, angle, gp = gpar(),
   # the same height regardless of whether the text actually contains letters with
   # descenders or not. The same happens automatically with ascenders already.
-  descent <- font_descent(gp$fontfamily, gp$fontface, gp$fontsize, gp$cex)
+  descent   <- font_descent(gp$fontfamily, gp$fontface, gp$fontsize, gp$cex)
 
   # Use trigonometry to calculate grobheight and width for rotated grobs. This is only
   # exactly correct when vjust = 1. We need to take the absolute value so we don't make
   # the grob smaller when it's flipped over.
-  text_height <- unit(1, "grobheight", text_grob) + abs(cos(angle[1] / 180 * pi)) * descent
-  text_width <- unit(1, "grobwidth", text_grob) + abs(sin(angle[1] / 180 * pi)) * descent
+  rad <- (angle[1] %% 360) / 180 * pi
+  x_descent <- abs(sin(rad)) * descent
+  y_descent <- abs(cos(rad)) * descent
 
-  if (isTRUE(debug)) {
-    children <- gList(
-      rectGrob(gp = gpar(fill = "cornsilk", col = NA)),
-      pointsGrob(x, y, pch = 20, gp = gpar(col = "gold")),
-      text_grob
-    )
-  } else {
-    children <- gList(text_grob)
-  }
-
-  list(
-    text_grob = children,
-    text_height = text_height,
-    text_width = text_width
-  )
-}
-
-#' Add margins
-#'
-#' Given a text grob, `add_margins()` adds margins around the grob in the
-#' directions determined by `margin_x` and `margin_y`.
-#'
-#' @param grob A gList containing a grob, such as a text grob
-#' @param height,width Usually the height and width of the text grob. Passed as
-#'   separate arguments from the grob itself because in the special case of
-#'   facet strip labels each set of strips should share the same height and
-#'   width, even if the labels are of different length.
-#' @inheritParams titleGrob
-#'
-#' @noRd
-add_margins <- function(grob, height, width, margin = NULL,
-                        gp = gpar(), margin_x = FALSE, margin_y = FALSE) {
+  # Set text size to actual size including descenders
+  width  <- unit(1, "grobwidth",  grob) + x_descent
+  height <- unit(1, "grobheight", grob) + y_descent
 
+  # Resolve margin
   if (is.null(margin)) {
     margin <- margin(0, 0, 0, 0)
   }
+  margin_x <- isTRUE(margin_x)
+  margin_y <- isTRUE(margin_y)
 
-  if (margin_x && margin_y) {
-    widths <- unit.c(margin[4], width, margin[2])
-    heights <- unit.c(margin[1], height, margin[3])
+  # Initialise new values for position and dimensions
+  new_x <- new_y <- new_width <- new_height <- NULL
 
-    vp <- viewport(
-      layout = grid.layout(3, 3, heights = heights, widths = widths),
-      gp = gp
-    )
-    child_vp <- viewport(layout.pos.row = 2, layout.pos.col = 2)
-  } else if (margin_x) {
-    widths <- unit.c(margin[4], width, margin[2])
-    vp <- viewport(layout = grid.layout(1, 3, widths = widths), gp = gp)
-    child_vp <- viewport(layout.pos.col = 2)
+  # Calculate new x/width
+  if (margin_x) {
+    new_width  <- unit.c(margin[4], width, margin[2])
```

Comment:
another double space :-)
```suggestion
    new_width <- unit.c(margin[4], width, margin[2])
```

Reply:
Thanks for catching these; they are 'scars' of code alignment that I reversed later :')